src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -28,6 +29,10 @@
         "type": (str, "type of service"),
         "id": (str, "Service Id"),
     }], "Services related to the host"),
+    "service_instances": ({
+        "type": (str, "type of service"),
+        "count": (int, "Number of service instances"),
+    }, "Service instances related to the host"),
```

Comment:
I don't think this is correct. For this to be the case, it should be corrected (missing `[]`):
```suggestion
    "service_instances": ([{
        "type": (str, "type of service"),
        "count": (int, "Number of service instances"),
    }], "Service instances related to the host"),
```

However, this PR is generating this:
```json
{
  "mgr": 1,
  "mon": 3
}
```
While, according to the above schema, it should rather be:
```json
[
  {
    "type": "mgr",
    "count": 1
  },
  {
    "type": "mon",
    "count": 3
  }
]
```
Alternatively, we should fix how `service_instance` is defined above to produce an `object` with either:
```
"additionalProperties": { "type": "integer" }`
```
Or (this would be probably hard to maintain, since new services have to update this):
```
"patternProperties": {
    "^(mgr|mon|osd|rgw)$": { "type": "integer" },
```

Now I'm further reflecting on this, I think that it's good to keep that data in the front-end, but not as it was before. Let me explain myself:
* `service_instance` is derived information from the `services` field.
* We should provide as much info as we can from the back-end with a couple of exceptions:
  * avoid redundant information (as this case, `service_instance` can be easily obtained from `services`),
  * avoid "processed" information (since information presentation is a concern of the front-end layer).

So my suggestion would be to use an Angular Pipe to convert the `services` into a counter-like structure.

Reply:
I updated the schema. And regarding the second comment, I think its best for now to keep this in the backend because instead of a single source of truth we are relying on two (ceph services and orch daemons). And I guess both of them are needed atleast for now. So if we rely on the information derived from the services, it would be misleading when the orch is present.